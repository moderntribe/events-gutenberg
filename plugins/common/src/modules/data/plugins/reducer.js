/**
 * Internal dependencies
 */
import { types } from '@moderntribe/common/data/plugins';

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_PLUGIN:
			return {
				...state,
				[ action.payload.name ]: true,
			};
		case types.REMOVE_PLUGIN:
			return {
				...state,
				[ action.payload.name ]: false,
			};
		default:
			return state;
	}
};
