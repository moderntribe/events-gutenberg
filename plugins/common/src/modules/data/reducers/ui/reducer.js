/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { accordion } from './reducers';

export default combineReducers( {
	accordion,
} );
