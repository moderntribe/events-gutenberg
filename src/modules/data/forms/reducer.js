/**
 * Internal dependencies
 */
import * as types from './types';
import { form } from './reducers';

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_FORM:
		case types.CLEAR_FORM:
		case types.SET_FORM_FIELDS:
		case types.CREATE_FORM_DRAFT:
		case types.EDIT_FORM_ENTRY:
		case types.SUBMIT_FORM:
			return {
				...state,
				[ action.payload.id ]: form( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};
