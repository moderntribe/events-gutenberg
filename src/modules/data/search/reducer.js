/**
 * Internal dependencies
 */
import * as types from './types';
import { EVENT } from 'editor/post-types';

export const DEFAULT_STATE = {
	term: '',
	results: [],
	page: 1,
	totalPages: 0,
	loading: false,
	postType: EVENT,
};

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK:
			return {
				...state,
				[ action.payload.id ]: DEFAULT_STATE,
			};
		case types.CLEAR_BLOCK:
			return {
				...state,
				[ action.payload.id ]: {
					...DEFAULT_STATE,
					postType: state[ action.payload.id ].postType
				},
			};
		case types.SET_TERM:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					term: action.payload.term,
				},
			};
		case types.SET_RESULTS:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					results: action.payload.results,
				},
			};
		case types.ADD_RESULTS:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					results: [ ...state[ action.payload.id ], ...action.payload.results ],
				},
			};
		case types.SET_PAGE:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					page: action.payload.results,
				},
			};
		case types.SET_TOTAL_PAGES:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					totalPages: action.payload.totalPages,
				},
			};
		case types.SET_LOADING:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					loading: action.payload.loading,
				},
			};
		case types.SET_POST_TYPE:
			return {
				...state,
				[ action.payload.id ]: {
					...state[ action.payload.id ],
					postType: action.payload.results,
				},
			};
		default:
			return state;
	}
};
