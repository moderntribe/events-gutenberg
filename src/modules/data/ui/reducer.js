/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	dashboardDateTimeOpen: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DASHBOARD_DATE_TIME:
			return {
				...state,
				dashboardDateTimeOpen: action.payload.open,
			};
		case types.TOGGLE_DASHBOARD_DATE_TIME:
			return {
				...state,
				dashboardDateTimeOpen: ! state.dashboardDateTimeOpen,
			};
		default:
			return state;
	}
};
