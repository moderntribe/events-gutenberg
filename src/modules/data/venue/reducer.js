/**
 * External imports
 */
import { isEmpty, get } from 'lodash';
import { dispatch } from '@wordpress/data';

const { apiRequest } = wp;

/**
 * Internal imports
 */
import { STORE_NAME, POST_TYPE } from './index';

export const getAddress = ( meta = {} ) => {
	if ( isEmpty( meta ) ) {
		return {};
	}

	const {
		_VenueAddress,
		_VenueCity,
		_VenueProvince,
		_VenueZip,
		_VenueCountry,
	} = meta;

	return {
		street: _VenueAddress,
		city: _VenueCity,
		province: _VenueProvince,
		zip: _VenueZip,
		country: _VenueCountry,
	};
};

export const getCoordinates = ( meta = {} ) => {
	const { _VenueLat, _VenueLng } = meta;
	const lat = parseFloat( _VenueLat );
	const lng = parseFloat( _VenueLng );

	return {
		lat: isNaN( lat ) ? null : lat,
		lng: isNaN( lng ) ? null : lat,
	};
};

export const createDraft = ( state, fields ) => {
	apiRequest( {
		path: `/wp/v2/${ POST_TYPE }`,
		method: 'POST',
		data: fields,
	} ).then( ( body ) => {
		const { id } = body;
		dispatch( STORE_NAME ).registerVenue( id );
		dispatch( STORE_NAME ).setDetails( id, { ...body, volatile: true } );
	} );

	return {
		...state,
		loading: true,
		submit: true,
	};
};

export const editDraft = ( state, id, fields ) => {
	apiRequest( {
		path: `/wp/v2/${ POST_TYPE }/${ id }`,
		method: 'PUT',
		data: fields,
	} ).then( ( body ) => {
		dispatch( STORE_NAME ).setDetails( body.id, { ...body, volatile: true } );
	} );

	return {
		...state,
		loading: true,
		submit: true,
	};
};

export const removeDraft = ( state, id ) => {
	apiRequest( {
		path: `/wp/v2/${ POST_TYPE }/${ id }`,
		method: 'DELETE',
	} ).then( () => {
		dispatch( STORE_NAME ).clear();
	} );

	return {
		...state,
		loading: true,
	};
};

export const setDetails = ( prevState, id, details ) => {
	const meta = get( details, 'meta', {} );
	return {
		...prevState,
		id,
		details,
		address: getAddress( meta ),
		coordinates: getCoordinates( meta ),
		edit: false,
		create: false,
		loading: false,
		submit: false,
	};
};
