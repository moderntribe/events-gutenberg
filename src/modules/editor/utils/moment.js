/**
 * External dependencies
 */
import { isString } from 'lodash';
import moment from 'moment/moment';
import { FORMATS } from './date';
import { replaceWithObject } from './string';

/**
 * Make sure the format provided matches the spec used by moment.js
 *
 * @param {string} format The format to be converted to a moment format
 * @returns {string} return a moment.js valid format
 */
export function toFormat( format ) {
	const strtr = ( str, pairs ) => {
		const substrs = Object.keys( pairs ).map( escapeRegExp );
		return str.split( RegExp( `(${ substrs.join( '|' ) })` ) )
			.map( part => pairs[ part ] || part )
			.join( '' );
	};

	const replacements = {
		d: 'DD',
		D: 'ddd',
		j: 'D',
		l: 'dddd',
		N: 'E',
		S: 'o',
		w: 'e',
		z: 'DDD',
		W: 'W',
		F: 'MMMM',
		m: 'MM',
		M: 'MMM',
		n: 'M',
		t: '', // no equivalent
		L: '', // no equivalent
		o: 'YYYY',
		Y: 'YYYY',
		y: 'YY',
		a: 'a',
		A: 'A',
		B: '', // no equivalent
		g: 'h',
		G: 'H',
		h: 'hh',
		H: 'HH',
		i: 'mm',
		s: 'ss',
		u: 'SSS',
		e: 'zz', // deprecated since version 1.6.0 of moment.js
		I: '', // no equivalent
		O: '', // no equivalent
		P: '', // no equivalent
		T: '', // no equivalent
		Z: '', // no equivalent
		c: '', // no equivalent
		r: '', // no equivalent
		U: 'X',
	};

	return strtr( format, replacements );
}

/**
 * Round the time of a moment object if the minutes on the date is lower than 30 will set to 0 if
 * is greater will se 30 so is either 30 or 0.
 *
 * @param {Moment} date Make sure the date is rounded between 0 or 30 minutes
 * @returns {Moment} A moment object
 */
export function roundTime( date ) {
	if ( ! ( date instanceof moment ) ) {
		return date;
	}

	let minutes = date.minute();
	if ( minutes >= 30 ) {
		minutes = ( minutes % 30 );
	}

	return date
		.clone()
		.subtract( minutes, 'm' )
		.seconds( 0 );
}

/**
 * Convert a Date() object into a Moment.js object avoiding warnings of different formats
 * used by Date
 *
 * @param {(Date|moment|string)} date The date to be converted.
 * @param {string} format The format of the data to be used
 * @returns {Moment} A moment object
 */
export function toMoment( date, format = FORMATS.WP.datetime ) {
	if ( date instanceof moment ) {
		return moment( date );
	} else if ( date instanceof Date ) {
		return toMomentFromDate( date );
	} else if ( isString( date ) ) {
		return moment( date, toFormat( format ) );
	}

	return moment();
}

function toMomentFromDate( date ) {
	if ( ! ( date instanceof Date ) ) {
		throw new Error( 'Make sure your date is an instance of Date' );
	}

	const year = date.getFullYear();
	const month = date.getMonth();
	const day = date.getDate();

	return moment()
		.year( year )
		.month( month )
		.date( day )
		.startOf( 'day' );
}

/**
 * Replace the date of a moment object with another date from another moment object
 *
 * @param {Moment} original The moment object where the date is going to be replaced
 * @param {Moment} replaced The moment object where the date to be used to replace is located
 * @returns {Moment} A moment object where the date is replaced
 */
export function replaceDate( original, replaced ) {
	if ( ! ( original instanceof moment ) || ! ( replaced instanceof moment ) ) {
		throw new Error( 'Make sure your values are instances of moment' );
	}

	return original
		.year( replaced.year() )
		.month( replaced.month() )
		.date( replaced.date() );
}

/**
 * Set time in seconds to a moment object
 *
 * @param {Moment} original The original moment where the date is going to be set
 * @param {number} seconds Amount of seconds to be set to the moment object.
 * @returns {Moment} A moment object with the new date
 */
export function setTimeInSeconds( original, seconds = 0 ) {
	if ( ! ( original instanceof moment ) ) {
		throw new Error( 'Make sure your values are instances of moment' );
	}

	if ( seconds < 0 ) {
		return original;
	}

	return original
		.startOf( 'day' )
		.seconds( seconds || original.seconds() );
}

/**
 * Total seconds of a current date from moment
 *
 * @param {Moment} date The date to compare on the current day
 * @returns {int} Total of seconds from start of the day to the current moment,
 */
export function totalSeconds( date ) {
	if ( ! date || ! ( date instanceof moment ) ) {
		return 0;
	}

	return date.diff( moment( date ).startOf( 'day' ), 'seconds' );
}

/**
 * Convert a moment object into a WP date time format
 *
 * @param {Moment} date A moment date object
 * @returns {string} A date time format
 */
export function toDateTime( date ) {
	return date.format( toFormat( FORMATS.WP.datetime ) );
}

export function toDate( date ) {
	return date.format( toFormat( FORMATS.WP.date ) );
}

export function toDateNoYear( date ) {
	return date.format( toFormat( FORMATS.WP.dateNoYear ) );
}

export function toTime( date ) {
	return date.format( toFormat( FORMATS.WP.time ) );
}

export function toDatePicker( date = moment(), format = 'YYYY-MM-DDTHH:mm:ss' ) {
	return date.format( format );
}

/**
 * Test if the current start and end date are happening on the same day.
 *
 * @param {Moment} start The start of the event
 * @param {(Moment|String)} end The end date of the event
 * @returns {boolean} if the event is happening on the same day
 */
export function isSameDay( start, end ) {
	return moment( start ).isSame( end, 'day' );
}
