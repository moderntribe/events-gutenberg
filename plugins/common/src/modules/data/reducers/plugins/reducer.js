/**
 * External dependencies
 */
import { omit } from 'lodash';

/**
 * Internal dependencies
 */
import { types } from '@moderntribe/common/data/reducers/plugins';

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_PLUGIN:
			return {
				...state,
				[ action.payload.name ]: true,
			};
		case types.REMOVE_PLUGIN:
			return omit( state, action.payload.name );
		default:
			return state;
	}
};
