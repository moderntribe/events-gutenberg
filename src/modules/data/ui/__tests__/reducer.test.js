/**
 * External dependencies
 */
import moment from 'moment';

/**
 * Internal dependencies
 */
import reducer, { actions } from 'data/ui';

describe( '[STORE] - UI reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toMatchSnapshot();
	} );

	it( 'Should Set the Dashboard date time by open it', () => {
		const current = reducer( {}, actions.openDashboard() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should set the Dashboard date time by closing it', () => {
		const current = reducer( {}, actions.closeDashboard() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should Toggle Dashboard date time', () => {
		const current = reducer( { dashboardDateTimeOpen: false }, actions.toggleDashboard() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should Set visibility month', () => {
		const visibleMonth = moment().startOf( 'month' ).toDate();
		const current = reducer( {}, actions.setVisibleMonth( visibleMonth ) );
		const expected = {
			visibleMonth,
		};
		expect( current ).toEqual( expected );
	} );
} );
