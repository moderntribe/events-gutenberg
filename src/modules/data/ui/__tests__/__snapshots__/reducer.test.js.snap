// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - UI reducer Should Set the Dashboard date time by open it 1`] = `
Object {
  "dashboardDateTimeOpen": true,
}
`;

exports[`[STORE] - UI reducer Should Toggle Dashboard date time 1`] = `
Object {
  "dashboardDateTimeOpen": true,
}
`;

exports[`[STORE] - UI reducer Should set the Dashboard date time by closing it 1`] = `
Object {
  "dashboardDateTimeOpen": false,
}
`;
